daemon/src/ipc_server.rs

```diff
@@ -120,6 +120,13 @@ fn handle_message(buffer: &mut String, ustream: UnixStream, wpaperd: &mut Wpaper
 
             IpcResponse::Ok
         }),
+
+        IpcMessage::ReloadConfig => {
+            wpaperd.reload_config().map(|_| IpcResponse::Ok)?;
+            // Force update
+            wpaperd.surfaces.iter_mut().for_each(|sur| sur.next_image());
```

Comment:
There is no need to call this function. When there is a message on the IPC, the main loop will wake up and check if the config has changed, applying all the modification needed.